static/js/hashauth/hashauth.js

```diff
@@ -0,0 +1,133 @@
+(function () {
+	var apisecret = '';
+	var storeapisecret = false;
+	var	apisecrethash = null;
+	var authenticated = false;
+	
+	function verifyAuthentication(next) {
+		var xhr = new XMLHttpRequest();
+		xhr.open('GET', '/api/v1/verifyauth', true);
+		xhr.setRequestHeader('Content-Type', 'application/json; charset=UTF-8');
+		xhr.setRequestHeader('api-secret', apisecrethash);
+		xhr.onload = function () {
+//console.log(xhr);
+			if (xhr.statusText=='OK') {
+				authenticated = true;
+				console.log('Authentication passed.');
+				next(true);
+			} else {
+				console.log('Authentication failed.');
+				removeAuthentication();
+				next(false);
+			}
+		}
+		xhr.send();
+	}
+	function initAuthentication() {
+		apisecrethash = localStorage.getItem('apisecrethash');
+		verifyAuthentication(function () {
+			$('#authentication_placeholder').html(inlineCode());
+		});
+	}
+	
+	function removeAuthentication(event) {
+		localStorage.removeItem('apisecrethash');
+		apisecret = null;
+		apisecrethash = null;
+		authenticated = false;
+		$('#authentication_placeholder').html(inlineCode());
+		if (event) event.preventDefault();
+		return false;
+	}
+	
+	function requestAuthentication(event) {
+		$( "#requestauthenticationdialog" ).dialog({
+			  width: 500
+			, height: 240
+			,  buttons: [
+				{ text: translate("Update"),
+					click: function() {
+						apisecret = $('#apisecret').val();
+						storeapisecret = $('#storeapisecret').is(':checked');
+						if (apisecret.length < 12) {
+							alert(translate('Too short API secret'));
+							$( this ).dialog( "close" );
+							return false;
+						} else {
+							var shaObj = new jsSHA(apisecret, "TEXT");
+							apisecrethash = shaObj.getHash("SHA-1", "HEX");
+							var dialog = this;
+							verifyAuthentication( function(isok) {
+								if (isok) {
+									if (storeapisecret) {
+										localStorage.setItem('apisecrethash',apisecrethash);
+									}
+									$('#authentication_placeholder').html(inlineCode());
+								} else {
+									alert(translate('Wrong API secret'));
+								}
+								$( dialog ).dialog( "close" );
+							});
+							return false;
+						}
+					}
+						
+				},
+				{ text: translate("Remove"),
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769498829/issues/source?bid=2452374&fileBranchId=2459081#l76)